samza-core/src/main/java/org/apache/samza/coordinator/staticresource/StaticResourceJobCoordinator.java

```diff
@@ -212,6 +225,21 @@ void doSetLoggingContextConfig(Config config) {
     LoggingContextHolder.INSTANCE.setConfig(config);
   }
 
+  /**
+   * Sets up {@link DiagnosticsManager} for the job coordinator.
+   */
+  private void setUpDiagnostics(JobModel jobModel) {
+    JobConfig jobConfig = new JobConfig(this.config);
+    String jobName = jobConfig.getName().orElseThrow(() -> new ConfigException("Missing job name"));
+    // TODO SAMZA-2705: construct execEnvContainerId for diagnostics
+    Optional<DiagnosticsManager> diagnosticsManager = buildDiagnosticsManager(jobName, jobConfig.getJobId(), jobModel,
+        CoordinationConstants.JOB_COORDINATOR_CONTAINER_NAME, Optional.empty(), config);
+    if (diagnosticsManager.isPresent()) {
+      diagnosticsManager.get().start();
+      this.currentDiagnosticsManager = diagnosticsManager;
```

Comment:
why not start the diagnosticsManager along the `coordinatorCommunication` after your setup diagnostics?
It will be more readable and relatable since we anyways `quietlyStop` if present as part of the `start()` method above.

Reply:
I moved a few things around in order to group together the assignment of instance variables and then group together the `start` lifecycle calls.